curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/643c565936a10109b3f59326.md

```diff
@@ -0,0 +1,253 @@
+---
+id: 643c565936a10109b3f59326
+title: Step 9
+challengeType: 0
+dashedName: step-9
+---
+
+# --description--
+
+To see your changes take effect as you write your `showLatestPosts()` function, call it at the end of your `fetchData()` function. Pass `data` as the argument.
+
+# --hints--
+
+You should call `showLatestPosts()` at the end of your `try` block inside the `fetchData` function.
+
+```js
+assert.match(code, /const\s*fetchData\s*=\s*async\s*\(\)\s*=>\s*{\s*try{\s*const\s*res\s*=\s*await\s*fetch\(FORUM_LATEST\);?\s*const\s*data\s*=\s*await\s*res\.json\(\);?\s*console\.log\(data\);?\s*showLatestPosts\(.*\);??\s*}\s*catch\(err\)\s*{\s*console\.log\(err\);?\s*}\s*}/);
```

Comment:
I am unable to pass this test with the correct code
<img width="875" alt="Screenshot 2023-11-16 at 12 33 43 PM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/6c85c0da-c559-4357-ba19-64ab33c70d6e">

Reply:
done